From cb3ef99c1553565e1dc0301ccd5c1c0fa2d15c15 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Fri, 31 Dec 2021 17:56:14 +0100
Subject: [PATCH] arm64: dts: ipq8074: add CPU clock

Now that CPU clock is exposed and can be controlled, add the necessary
properties to the CPU nodes.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq8074.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
@@ -5,6 +5,7 @@
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/clock/qcom,gcc-ipq8074.h>
+#include <dt-bindings/clock/qcom,apss-ipq.h>
 #include <dt-bindings/clock/qcom,ipq8074-cmn-pll.h>
 
 / {
@@ -45,6 +46,8 @@
 			reg = <0x0>;
 			next-level-cache = <&l2_0>;
 			enable-method = "psci";
+			clocks = <&apcs_glb APCS_ALIAS0_CORE_CLK>;
+			clock-names = "cpu";
 		};
 
 		cpu1: cpu@1 {
@@ -53,6 +56,8 @@
 			enable-method = "psci";
 			reg = <0x1>;
 			next-level-cache = <&l2_0>;
+			clocks = <&apcs_glb APCS_ALIAS0_CORE_CLK>;
+			clock-names = "cpu";
 		};
 
 		cpu2: cpu@2 {
@@ -61,6 +66,8 @@
 			enable-method = "psci";
 			reg = <0x2>;
 			next-level-cache = <&l2_0>;
+			clocks = <&apcs_glb APCS_ALIAS0_CORE_CLK>;
+			clock-names = "cpu";
 		};
 
 		cpu3: cpu@3 {
@@ -69,6 +76,8 @@
 			enable-method = "psci";
 			reg = <0x3>;
 			next-level-cache = <&l2_0>;
+			clocks = <&apcs_glb APCS_ALIAS0_CORE_CLK>;
+			clock-names = "cpu";
 		};
 
 		l2_0: l2-cache {
